### PR TITLE
Use a unified Options object

### PR DESCRIPTION
**Pull request by ddfisher**
- Move `build_flags` into Options.
- Parse CLI flags onto Options directly when possible to avoid repetition.
- Pass around Options objects instead of unpacking them.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [flake8]
-max-line-length = 99
+max-line-length = 120
```
Comment (gvanrossum) [inline setup.cfg]: Oh no you don't!

**Comment by gvanrossum** [comment]
I really don't like bumping max line width. But I like everything else. Great diffstats, too!

**Comment by ddfisher** [comment]
Unbumped max line length.  <_<

Woo net negative LOC!

**Comment by gvanrossum** [comment]
Whee!

**Pull request merged**